test/test-apimatic-core/union_type_tests/one_of_any_of_test.rb

```diff
@@ -0,0 +1,374 @@
+require 'minitest/autorun'
+require 'apimatic_core'
+
+require_relative '../../test-helper/models/morning'
+require_relative '../../test-helper/models/evening'
+require_relative '../../test-helper/models/month_number_enum'
+require_relative '../../test-helper/models/month_name_enum'
+
+class OneOfAnyOfTest < Minitest::Test
+  include CoreLibrary, TestComponent
+
+  def setup
+  end
+
+  def teardown
+  end
+
+  def test_primitive_one_of_any_of
+    _nested_any_of_one_of = AnyOf.new(
+      [
+        OneOf.new(
+          [
+            LeafType.new(String),
+            LeafType.new(Integer)
+          ]
+        ),
+        LeafType.new(TrueClass),
+        LeafType.new(FalseClass)
+      ]
+    )
+    _nested_any_of_one_of.validate('string')
+    assert _nested_any_of_one_of.is_valid
+  end
+
+  def test_invalid_primitive_one_of_any_of
+    _nested_any_of_one_of = AnyOf.new(
+      [
+        OneOf.new(
+          [
+            LeafType.new(String),
+            LeafType.new(String)
+          ]
+        ),
+        LeafType.new(TrueClass),
+        LeafType.new(FalseClass)
+      ]
+    )
+    assert_raises AnyOfValidationException do
```

Comment:
Blocker: The assertion against the error messages that the exception has thrown is missing. I'd recommend to see the tests in python core library.